packages/react/src/runtime/__tests__/style-ssr.test.tsx

```diff
@@ -43,12 +43,19 @@ describe('<Style />', () => {
           `._g1234567:visited{ color: grey; }`,
           `._h1234567:focus-visible{ color: white; }`,
           `._i1234567:focus-within{ color: black; }`,
+          `._j1234567{ border: none; }`,
+          `._j1234567{ border-top: 1px solid transparent; }`,
+          `._j1234567{ border-top-color: red; }`,
+          `._j1234567:focus{ border-top-color: transparent; }`,
+          `._j1234567:focus{ border-color: blue; }`,
+          `._j1234567:focus{ border: 1px solid; }`,
         ]}
       </Style>
     );
 
+    // WARNING: This is wrong, the `focus` border properties are different than without focus, borders would be indeterministic.
```

Comment:
If this is required, all "bucket sorting" needs to be drastically rewritten to be nested.

Right now it's flat:
```css
.abc { border: none }
.abc:link { border: none }
.abc:visited { border: none }
…
.abc:active { border: none }
@media (…) { .abc { border: none } }
```

But this doesn't support `@media (…)` being sorted internally or `.abc:focus { … }` being sorted against each other.

This would need deeper nested bucketing:
```css
.abc { border: none }
.abc{ border-top: none }
.abc:link { border: none }
.abc:link { border-top: none }
@media (…) {
  .abc { border: none }
  .abc { border-top: none }
  .abc:link { border: none }
  .abc:link { border-top: none }
}
```

And I'm not exactly sure if that's even feasible for `@media` queries either…